src/power_grid_model/core/power_grid_model.py

```diff
@@ -200,10 +200,14 @@ def _construct_output(
 
     @staticmethod
     def _options(**kwargs) -> Options:
-        if "calculation_method" in kwargs:
-            calculation_method = kwargs["calculation_method"]
-            if isinstance(calculation_method, str):
-                kwargs["calculation_method"] = CalculationMethod[calculation_method]
+        def replace_to_enum(key: str, type_: Type):
+            if key in kwargs:
+                value = kwargs[key]
+                if isinstance(value, str):
+                    kwargs[key] = type_[value]
```

Comment:
please double-check the sonar cloud code smell: You may need to replace this with `Type[IntEnum]` or otherwise need to add an additional check that it `issubclass(type_, IntEnum)`